packages/meditrak-server/src/tests/routes/GETSurveyResponses/assertSurveyResponsesPermission.test.js

```diff
@@ -0,0 +1,144 @@
+} from '@tupaia/database';
+import { getModels } from '../../getModels';
+import {
+  filterSurveyResponsesByPermissions,
+  assertSurveyResponsePermissions,
+} from '../../../routes/GETSurveyResponses/assertSurveyResponsePermissions';
+
+describe('Permissions checker for GETSurveyResponses', async () => {
+  const DEFAULT_POLICY = {
+    VU: ['Admin', 'Donor'],
+    LA: ['Admin'],
+  };
+
+  const models = getModels();
+  let surveyResponses = [];
+  const laosAdminResponseId = generateTestId();
+
+  before(async () => {
+    const adminPermissionGroup = await findOrCreateDummyRecord(models.permissionGroup, {
+      name: 'Admin',
+    });
+    const donorPermissionGroup = await findOrCreateDummyRecord(models.permissionGroup, {
+      name: 'Donor',
+    });
+    const vanuatuCountry = await findOrCreateDummyRecord(models.country, { code: 'VU' });
+    const laosCountry = await findOrCreateDummyRecord(models.country, { code: 'LA' });
+    const vanuatuEntity = await findOrCreateDummyRecord(models.entity, {
+      id: generateTestId(),
+      code: 'VU_TEST',
+      country_code: vanuatuCountry.code,
+    });
+    const laosEntity = await findOrCreateDummyRecord(models.entity, {
+      id: generateTestId(),
+      code: 'LA_TEST',
+      country_code: laosCountry.code,
+    });
+
+    await buildAndInsertSurveys(models, [
+      {
+        id: generateTestId(),
+        code: 'TEST_SURVEY_1',
+        name: 'Test Survey 1',
+        permission_group_id: adminPermissionGroup.id,
+      },
+      {
+        id: generateTestId(),
+        code: 'TEST_SURVEY_2',
+        name: 'Test Survey 2',
+        permission_group_id: donorPermissionGroup.id,
+      },
+    ]);
+
+    const surveyResponseModels = await buildAndInsertSurveyResponses(models, [
+      {
+        id: generateTestId(),
+        surveyCode: 'TEST_SURVEY_1',
+        entityCode: vanuatuEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: [],
+      },
+      {
+        id: laosAdminResponseId,
+        surveyCode: 'TEST_SURVEY_1',
+        entityCode: laosEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: [],
+      },
+      {
+        id: generateTestId(),
+        surveyCode: 'TEST_SURVEY_2',
+        entityCode: vanuatuEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: [],
+      },
+      {
+        id: generateTestId(),
+        surveyCode: 'TEST_SURVEY_2',
+        entityCode: laosEntity.code,
+        submission_time: '2020-01-31T09:00:00Z',
+        answers: [],
+      },
+    ]);
+    surveyResponses = surveyResponseModels.map(srm => srm.surveyResponse);
+  });
+
+  describe('filterSurveyResponsesByPermissions()', async () => {
+    it('Should filter any survey responses that users do not have access to to the survey of', async () => {
+      //Remove the permission of VU to have insufficient permissions to some surveys
+      const policy = {
+        // VU: ['Admin', 'Donor'],
```

Comment:
I did the same in my code and Edwin suggested me to changed this to

```suggestion
        VU: [/*'Admin'*/ 'Donor'],
```
Which I think makes it a better test case because it can check the case of having at least some permissions to the survey country but not sufficient for the permission group of the survey.

Same goes to other test cases

Reply:
That situation is already being tested in this test.

VU has no permissions.
LA has permissions for one of the surveyResponses but not the other.

I thought that was a reasonable spread to show the filtering working correctly.